addons/ondevice-controls/src/types/Boolean.tsx

```diff
@@ -3,40 +3,24 @@ import React from 'react';
 
 export interface BooleanProps {
   onChange: (value: boolean) => void;
-  knob: {
+  arg: {
     name: string;
     value: boolean;
   };
 }
 
-class BooleanType extends React.Component<BooleanProps, {}> {
-  static defaultProps = {
-    knob: {},
-    onChange: (value) => value,
-  };
-
-  static serialize = (value) => (value ? String(value) : null);
-
-  static deserialize = (value) => value === 'true';
-
-  onValueChange = () => {
-    const { onChange, knob } = this.props;
-    onChange(!knob.value);
-  };
-
-  render() {
-    const { knob } = this.props;
-
-    return (
-      <View style={styles.switch}>
-        <Switch testID={knob.name} onValueChange={this.onValueChange} value={knob.value} />
-      </View>
-    );
-  }
-}
+const BooleanType = ({ arg, onChange = (value) => value }: BooleanProps) => (
```

Comment:
Not sure about `= (value) => value`; is it necessary here? The typing suggests we expect an `onChange` always, so does it make sense to have a default value for it?

This situation is similar to https://github.com/storybookjs/react-native/pull/182/commits/a33b28179d3d2e23e6b26c0864cdd1ba4bc51d27#diff-7adb84faada991553c3c1a297f044b86d5b8abd6b1db5e09be8d16e66c91321cR31 , so added it just to play safe. But not sure 🤔 .

Reply:
Yeah right now I've mostly just been directly refactoring the knobs components however I think that the onchange default is not strictly necessary. To me its fine if you leave it or remove it so up to you. I expect that there will be some iterative improvements to controls. Currently it's more of a refactor of knobs but that will likely evolve.

Reply:
Thanks! Took it out, but let's restore it if we spot a reason to have it. Not a big deal either way, was just curious 🙂 .